core/domain/rights_manager.py

```diff
@@ -89,11 +114,25 @@ def _save_activity_rights(
     """
     activity_rights.validate()
 
+    # Ruling out the possibility of any other activity type.
+    assert (
+        activity_type in (
+            constants.ACTIVITY_TYPE_COLLECTION,
+            constants.ACTIVITY_TYPE_EXPLORATION
+        )
+    )
```

Comment:
```suggestion
    # Ruling out the possibility of any other activity type.
    assert (
        activity_type in (
            constants.ACTIVITY_TYPE_COLLECTION,
            constants.ACTIVITY_TYPE_EXPLORATION
        )
    )
```
Why is this needed? Can we create an else below with an error?

Reply:
ditto below

Reply:
A small discussion regarding this can be found here https://github.com/oppia/oppia/pull/15693#discussion_r919937597

The summary of the discussion is :
If we add an else case and throw an error, then that error is never executed, because in the whole file we are calling `_get_activity_rights` before calling this method and in `_get_activity_rights` we are already checking for this case.

https://github.com/oppia/oppia/blob/5f1b2e12f61f3daf750c44cc5dab58071ad582aa/core/domain/rights_manager.py#L471

Reply:
Okay!